src/main/modules/oidc/index.ts

```diff
@@ -20,32 +21,39 @@ export class Oidc {
 
     app.get(AuthUrls.CALLBACK, async (req: AppRequest, res: Response, next: NextFunction) => {
       const redisClient = req.app.locals?.redisClient;
-      const guid = req.query?.state;
-      if (redisClient && guid) {
-        redisClient.get(guid, (err: Error, typesOfClaim: string) => {
-          if (typesOfClaim) {
-            req.session.userCase.typeOfClaim = JSON.parse(typesOfClaim);
-          }
-          if (err) {
-            const error = new Error(err.message);
-            error.name = RedisErrors.FAILED_TO_RETREIVE;
-            if (err.stack) {
-              error.stack = err.stack;
-            }
-            return next(error);
-          }
-        });
-      } else {
-        const err = new Error(RedisErrors.CLIENT_NOT_FOUND);
-        err.name = RedisErrors.FAILED_TO_CONNECT;
-        return next(err);
-      }
 
-      if (typeof req.query.code === 'string') {
-        req.session.user = await getUserDetails(serviceUrl(res), req.query.code, AuthUrls.CALLBACK);
-        req.session.save(() => res.redirect(PageUrls.NEW_ACCOUNT_LANDING));
+      //Only go to new account landing page if it is a new user
+      if (req.query?.state) {
+        const guid = String(req.query?.state);
+        if (typeof req.query.code === 'string') {
+          req.session.user = await getUserDetails(serviceUrl(res), req.query.code, AuthUrls.CALLBACK);
+          req.session.save();
+        } else {
+          return res.redirect(AuthUrls.LOGIN);
+        }
+        if (!req.session?.user?.accessToken) {
+          return res.redirect(PageUrls.LIP_OR_REPRESENTATIVE);
+        }
+
+        if (redisClient && guid) {
+          getPreloginCaseData(redisClient, guid)
+            .then(caseType =>
+              createCase(caseType, req.session.user.accessToken, config.get('services.etSyaApi.host'))
+                .then(() => {
+                  return res.redirect(PageUrls.NEW_ACCOUNT_LANDING);
+                })
+                .catch(() => {
+                  //ToDo - needs to handle different error response
+                })
+            )
+            .catch(err => next(err));
+        } else {
+          const err = new Error(RedisErrors.CLIENT_NOT_FOUND);
```

Comment:
Should this else block only be run if there is no redisClient only?- rather than if there is no guid or no redis.